lightstep/sdk/metric/asyncinst.go

```diff
@@ -0,0 +1,59 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metric // import "github.com/lightstep/otel-launcher-go/lightstep/sdk/metric"
+
+import (
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/instrument/asyncfloat64"
+	"go.opentelemetry.io/otel/metric/instrument/asyncint64"
+	"github.com/lightstep/otel-launcher-go/lightstep/sdk/metric/internal/asyncstate"
+	"github.com/lightstep/otel-launcher-go/lightstep/sdk/metric/number"
+	"github.com/lightstep/otel-launcher-go/lightstep/sdk/metric/sdkinstrument"
+)
+
+type (
+	asyncint64Instruments   struct{ *meter }
+	asyncfloat64Instruments struct{ *meter }
+)
+
+func (i asyncint64Instruments) Counter(name string, opts ...instrument.Option) (asyncint64.Counter, error) {
```

Comment:
If the API itself supported generics, then we could do that here too. Because the return types are `asyncint64.Counter` and `asyncfloat64.Counter`, for example, it doesn't work.  If the API were to extend to use generics, then possibly.